### PR TITLE
[libpq] update to 12.0 and use vcpkg_configure_make

### LINKED ISSUE #8462: [libpq] update to 12.0 / improvements to vcpkg_configure_make
PostgreSQL 12.0 was just released and I am trying to get it to work with the new vcpkg_configure_make command. However psql needs a C99 compliant compiler, which cl is not. So I think I would need the possibility to use gcc with vcpkg_configure_cmake on Windows.

### PR DESCRIPTION
**Pull request by renshi-code**
will close #8462

### CODE REVIEW

**Comment by cenit** [comment]
This means that you can also remove the custom CMakeLists.txt?

**Comment by TheScarfix** [comment]
yes I hope to as soon as it works

**Comment by cenit** [comment]
I noted it only to not forget it. Since I just finished a tedious cleanup of many old and deprecated patches, I like to keep the repo clean now :)

File: ports/libpq/portfile.cmake [hunk 1]
```diff
@@ -5,28 +5,27 @@ if(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
 endif()
 
 vcpkg_download_distfile(ARCHIVE
-    URLS "https://ftp.postgresql.org/pub/source/v9.6.3/postgresql-9.6.3.tar.bz2"
-    FILENAME "postgresql-9.6.3.tar.bz2"
-    SHA512 97141972e154e6b0e756ee6a4e20f26e82022a9fd4c56305314a3a5567a79ece638e4ac3d513b46138737ae6bd27a098f30013a94767db151181aac9c01290a1
+    URLS "https://ftp.postgresql.org/pub/source/v12.0/postgresql-12.0.tar.bz2"
```
Comment (JackBoosY) [inline ports/libpq/portfile.cmake]: I think it should be set as a macro,

File: ports/libpq/portfile.cmake [hunk 2]
```diff
@@ -5,28 +5,27 @@ if(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
...
     ARCHIVE ${ARCHIVE}
 )
 
-file(COPY ${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt DESTINATION ${SOURCE_PATH})
-
-vcpkg_configure_cmake(
+vcpkg_configure_make(
     SOURCE_PATH ${SOURCE_PATH}
-    PREFER_NINJA
+    if(CMAKE_HOST_UNIX AND NOT CMAKE_HOST_APPLE)
```
Comment (JackBoosY) [inline ports/libpq/portfile.cmake]: Does it work?

Comment (TheScarfix) [reply ports/libpq/portfile.cmake]: looks like it doesn't

**Comment by Neumann-A** [comment]
please fix the title. there is no `change to vcpkg_configure_make` as far as I can see from the changes.

**Comment by cenit** [comment]
it was `vcpkg_configure_cmake`, now it is `vcpkg_configure_make`

**Comment by Neumann-A** [comment]
I read it as a change to the function/script `vcpkg_configure_make`.
Maybe it should be "and use vcpkg_configure_make" or "remove custom CMake build" or "use native build scripts".

**Comment by Neumann-A** [comment]
So I hacked the C99 check away (missing register keyword and the C equivalent to constexpr?) and fixed some Compiler/Linker flags but got stuck with the following error:

```
conftest.c
configure:8526: checking for non-GNU ld
configure:8559: result: no
configure:8562: error: no acceptable ld found in $PATH
```

If you then force in the portfile:
set(ENV{LD} "link.exe")
you get:

warnings:
```
cl : Command line warning D9035 : option 'o' has been deprecated and will be removed in a future release
cl : Command line warning D9002 : ignoring unknown option '-g'
cl : Command line warning D9002 : ignoring unknown option '--allow-multiple-definition'
cl : Command line warning D9002 : ignoring unknown option '--disable-auto-import'
cl : Command line warning D9002 : ignoring unknown option '-lz'
cl : Command line warning D9002 : ignoring unknown option '-lm'
```

```
configure:11813: result: no
configure:11823: error: zlib library not found
If you have zlib already installed, see config.log for details on the
failure.  It is possible the compiler isn't looking in the proper directory.
Use --without-zlib to disable zlib support.
```

I mean yeah you could patch the whole configure script on windows but it is probably easier to roll our own CMakeLists.txt file for libpq on Windows

Also from the config log some questionable checks:
`checking for C/C++ restrict keyword... no` 
which is a double check due to the C99 check ....
`checking for main in -lm... yes` 
because cl is just ignore -lm.....

**Comment by TheScarfix** [comment]
I'll try the MSBuild script later https://www.postgresql.org/docs/12/install-windows.html

**Comment by Neumann-A** [comment]
yeah that is probably the way to go. the files are in `src\tools\msvc` if you are looking for the scripts

**Comment by Neumann-A** [comment]
after reviewing the configure log you probably also need to explicitly pass the path of external dependencies like zlib to the configure script on linux to get consistent builds or the script will check if it can link -lz and if zlib is installed system wide (not the vcpkg version) it will catch that dependency and result in an inconsistent build.

**Comment by Neumann-A** [comment]
Furthermore if all configure options of libpq like zlib and libxml are features of libpq I need #8269 solved to make Qt work correctly in static builds.

**Comment by Neumann-A** [comment]
So got x86-windows to build without any features using the perl scripts.
https://github.com/Neumann-A/vcpkg/tree/libpq-12.0
you currently have to rename win_flex and win_bison to only bison and flex

**Comment by Neumann-A** [comment]
@TheScarfix: So I made the windows build work. I assume you want to have PR #8540 merged before working on the linux side?

**Pull request closed without merging**